io_mesh_w3d/w3x/export_w3x.py

```diff
@@ -14,65 +14,83 @@ def save(context, export_settings, data_context):
     root = create_root()
     includes = create_node(root, 'Includes')
 
+    dir = os.path.dirname(context.filepath) + os.path.sep
+
     if export_mode == 'M':
         if len(data_context.meshes) > 1:
             context.warning('Scene does contain multiple meshes, exporting only the first with export mode M!')
         data_context.meshes[0].header.container_name = ''
         data_context.meshes[0].header.mesh_name = data_context.container_name
         data_context.meshes[0].create(root)
 
-    elif export_mode == 'HM' or export_mode == 'HAM':
-        if export_mode == 'HAM' \
-                or not export_settings['use_existing_skeleton']:
-            data_context.hierarchy.header.name = data_context.container_name
-            data_context.hlod.header.hierarchy_name = data_context.container_name
+    elif export_mode == 'HM':
+        if export_settings['use_existing_skeleton'] or export_settings['individual_files']:
+            hierarchy_include = Include(type='all', source='ART:' + data_context.hierarchy.name() + '.w3x')
+            hierarchy_include.create(includes)
+        else:
             data_context.hierarchy.create(root)
 
-        elif not export_settings['individual_files']:
-            hierarchy_include = Include(type='all', source='ART:' + data_context.hierarchy.header.name + '.w3x')
-            hierarchy_include.create(includes)
+        if export_settings['individual_files']:
+            if not export_settings['use_existing_skeleton']:
+                path = dir + data_context.hierarchy.name() + context.filename_ext
+                context.info('Saving file :' + path)
+                write_struct(data_context.hierarchy, path)
 
-        dir = os.path.dirname(context.filepath) + os.path.sep
+        for texture in data_context.textures:
+            id = texture.split('.')[0]
```

Comment:
This doesn't seem correct. This breaks texture names with a dot

Reply:
Fixed